modes/vterm/evil-collection-vterm.el

```diff
@@ -62,6 +62,58 @@ also uses `evil-mode'."
                        "vterm"
                      "emacs"))))
 
+(defun vterm-evil-insert ()
+  (interactive)
+  (vterm-goto-char (point))
+  (call-interactively #'evil-insert))
+
+(defun vterm-evil-insert-line ()
+  (interactive)
+  (vterm-goto-char (vterm--get-prompt-point))
+  (call-interactively #'evil-insert))
+
+(defun vterm-evil-append ()
+  (interactive)
+  (vterm-goto-char (1+ (point)))
+  (call-interactively #'evil-append))
+
+(defun vterm-evil-append-line ()
+  (interactive)
+  (vterm-goto-char (vterm--get-end-of-line))
+  (call-interactively #'evil-append))
+
+(defun vterm-evil-delete ()
+  "Provide similar behavior as `evil-delete'."
+  (interactive)
+  (let ((inhibit-read-only t)
+        )
+    (cl-letf (((symbol-function #'delete-region) #'vterm-delete-region))
```

Comment:
Wondering what the purpose of this let is?

Reply:
Just to inhibit the read only buffer so that delete-region can work. I took it from the PR adding the functions to vterm here: https://github.com/akermu/emacs-libvterm/pull/455

I'm having trouble getting this one to work, it works in edebug, but strangely not when called/mapped during normal usage.

Reply:
maybe the `let` is not necessary , I tried it still words without `let`

Reply:
Resolving because I removed this